### ISSUE TITLE
Can't use EAC version of the game with the mod installed

### ISSUE THREAD

**Issue by 8492Ghosts** [WinterSquire/AlphaRing#90]
I installed the mod yesterday and was able to use the EAC-disabled version of the game and play splitscreen with another person, but this morning when I tried opening the version with EAC it gave the message in the screenshot. The .dll is installed to `C:\Program Files (x86)\Steam\steamapps\common\Halo The Master Chief Collection\MCC\Binaries\Win64`
![Screenshot 2024-12-24 110301](https://github.com/user-attachments/assets/616a3138-0126-4597-9e7f-4bce1bdac9ca)

**Comment by TheExpertNoob** [comment]
As expected, stated in README
https://github.com/WinterSquire/AlphaRing#installation
EAC off

**Comment by 8492Ghosts** [comment]
So is there no way to play the game with EAC on without uninstalling the .dll?

**Comment by TheExpertNoob** [comment]
Not at the moment as EAC sees it as an untrusted file.
The pain staking "workaround" is to rename it or delete it when switching.
Thankfully the following error box TELLS you which file it is. Delete it, when you want to play split-screen, re-add the dll and relaunch with EAC off.

**Comment by 8492Ghosts** [comment]
Ok, thank you.

**Comment by 8492Ghosts** [comment]
Ok, thank you.

**Comment by a-gos** [comment]
You could create a .bat similar to this one, and every time you run it you can choose which version of MCC to launch. With or without mods.

This the one I created, it may fail on your system and you will need to modify all the paths. 

GAME_DIR is the directory of the win64 folder of the game
MODS_DIR is a folder I created with the dlls
MODS_ENABLED_EXE is a shortcut to the exe to launch without EAC
MODS_DISABLED_EXE is a shortcut to the default exe

Also I have only used this with the Game Pass version

```powershell
@echo off
:: Paths
set "GAME_DIR=R:\XboxGames\Halo- The Master Chief Collection\Content\mcc\binaries\win64"
set "MODS_DIR=R:\XboxGames\Halo- The Master Chief Collection\Content\mcc\binaries\Mods"
set "MODS_ENABLED_EXE=R:\XboxGames\Halo- The Master Chief Collection\Halo MCC Anti-Cheat Disabled (Mods and Limited Services).lnk"
set "MODS_DISABLED_EXE=R:\XboxGames\Halo- The Master Chief Collection\Halo The Master Chief Collection.lnk"

:: Main Menu
cls
echo -----------------------------------
echo     Halo MCC Mod Management
echo -----------------------------------
echo [1] Enable Mods and launch Halo MCC (Modded)
echo [2] Disable Mods and launch Halo MCC (Unmodded)
echo [3] Exit
echo.

:: Get User Input
set /p "choice=Enter your choice [1/2/3]: "
:: Strip spaces and take the first character
set "choice=%choice: =%"
set "choice=%choice:~0,1%"
pause

:: Validate Input
if "%choice%"=="1" (
    move /Y "%MODS_DIR%\WTSAPI32.pdb" "%GAME_DIR%" || echo Failed to move WTSAPI32.pdb
    move /Y "%MODS_DIR%\WTSAPI32.dll" "%GAME_DIR%" || echo Failed to move WTSAPI32.dll
    start "" "%MODS_ENABLED_EXE%"
    pause
    exit
)

if "%choice%"=="2" (
    move /Y "%GAME_DIR%\WTSAPI32.pdb" "%MODS_DIR%" || echo Failed to move WTSAPI32.pdb
    move /Y "%GAME_DIR%\WTSAPI32.dll" "%MODS_DIR%" || echo Failed to move WTSAPI32.dll
    start "" "%MODS_DISABLED_EXE%"
    pause
    exit
)

if "%choice%"=="3" (
    pause
    exit
)

:: Invalid Input
pause
call "%~f0"

```

**Comment by 8492Ghosts** [comment]
Ok thanks, for now what I do is to play split screen I have the mod in the proper folder, and when I use the EAC version I just move it into a different folder.

**Comment by slice-the-idiot** [comment]
hi, steam deck user here

...this mod is working with eac on for some reason, haven't done any testing with splitscreen online, only solo matchmaking, is this just a byproduct of it being based on proton eac?

**Comment by vvdavided** [comment]
> hi, steam deck user here
> 
> ...this mod is working with eac on for some reason, haven't done any testing with splitscreen online, only solo matchmaking, is this just a byproduct of it being based on proton eac?

Same here on steam deck, I can launch the EAC enabled version and play online just fine while having the mod installed

**Comment by Squall-Leonhart** [comment]
the dll needs to be signed in order to be loaded with EAC.

**Comment by KohlJary** [comment]
Yeah can confirm, on Linux you can launch with anti-cheat enabled and the mod active and it loads fine, even popped an achievement. I didn't try online but I don't see any reason it wouldn't work.
